components/script/dom/element.rs

```diff
@@ -2245,6 +2245,20 @@ impl VirtualMethods for Element {
         }
 
         let doc = document_from_node(self);
+
+        if self.is_focusable_area() {
+            let doc = document_from_node(self);
+            let children = self.upcast::<Node>().traverse_preorder();
```

Comment:
I don't think we want to iterate over all the children here, since they should have their own bind_to_tree methods called already.

Reply:
If we only try to find the _`starting point`_ of focusable elements, I'll expect we don't need to traverse the children here and add them into the list.